File: Lab5/Task 1/CSharp/Line.cs
using System;
using System.Linq;
using System.Diagnostics.CodeAnalysis;


namespace CSharp
{
    internal class Line
    {
        protected double[][] Coordinates { get; }
        protected internal double Length { get; }

        protected Line(double[] start, double[] end)
        {
            this.Coordinates = new double[2][] {start, end};
            this.Length = GetLength();
        }

        private double GetLength()
        {
            var sum = 0.0;
            for (var i = 0; i < 3; i++)
            {
                sum += Math.Pow(Math.Abs(this.Coordinates[1][i]-this.Coordinates[0][i]), 2);
            }

            return Math.Sqrt(sum);
        }
    }
}

File: Lab5/Task 1/С++/С++/Line.cpp
#include "Line.hpp"
#include <math.h>

using namespace std;

double Line::CalculateLength() {
    double sum = 0.;
    for (int i = 0; i < 3; i++) {
        sum += pow(abs(this->Coordinates[1][i] - this->Coordinates[0][i]), 2);
    }
    return sqrt(sum);
}

Line::Line(double* start, double* end){
    double** Coordinates = new double*[2];
    Coordinates[0] = start;
    Coordinates[1] = end;
    this->Coordinates = Coordinates;
    this->Length = this->CalculateLength();
}

double Line::GetLength() { return this->Length; }
